Source/CourseContentPageViewController.swift

```diff
@@ -423,3 +439,22 @@ extension CourseContentPageViewController {
         moveInDirection(direction: .reverse)
     }
 }
+
+extension CourseContentPageViewController: UIGestureRecognizerDelegate {
+    public func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldRecognizeSimultaneouslyWith otherGestureRecognizer: UIGestureRecognizer) -> Bool {
+        return false
+    }
+    
+    public func gestureRecognizerShouldBegin(_ gestureRecognizer: UIGestureRecognizer) -> Bool {
+        if (gestureRecognizer == scrollViewPanGestureRecognzier)
+        {
+            let locationInView = gestureRecognizer.location(in: view)
+            if (locationInView.y > restrictedPaginationAreaStart && locationInView.y < restrictedPaginationAreaEnd)
+            {
+                return true
+            }
+            return false
+        }
+        return false
+    }
+}
```

Comment:
This can be better formatted and simplified as 

```suggestion
    public func gestureRecognizerShouldBegin(_ gestureRecognizer: UIGestureRecognizer) -> Bool {
        if (gestureRecognizer == scrollViewPanGestureRecognzier) {
            let locationInView = gestureRecognizer.location(in: view)
            if (locationInView.y > restrictedPaginationAreaStart && locationInView.y < restrictedPaginationAreaEnd) {
                return true
            }
        }
        return false
    }
```

Reply:
done